### PR TITLE
Fix TOSA broadcast and mixed precision tests

### PR DESCRIPTION
**Pull request by jamestcl-amd**
Fix the following TOSA tests:
- bf16xbf16_sub_elem_2d_broadcast_2d
- i8xi16_sub_elem

Add the following new TOSA tests:
- i16xi16_sub_elem_2d_broadcast_scalar (pass)
- i16xi16_sub_elem_2d_broadcast_1d_unit_dim (pass)
- bf16xbf16_sub_elem_2d_broadcast_scalar (xfail)

No new failures in the following regressions:
- test/Conversion/VectorToAIEVec
- test/Integration/Dialect
- test/unit_tests/aievec_tests

### CODE REVIEW

File: test/Integration/Dialect/TOSA/i8xi16_sub_elem/dut.cc [hunk 1]
```diff
@@ -0,0 +1,20 @@
+void dut(int8_t * restrict v1, int16_t * restrict v2, int32_t * restrict v3) {
```
Comment (linay-xsj) [inline test/Integration/Dialect/TOSA/i8xi16_sub_elem/dut.cc:1]: It seems that you haven't uploaded testbench.cc and defines.h file for this test.

File: test/Integration/Dialect/TOSA/bf16xbf16_sub_elem_2d_broadcast_scalar/bf16xbf16_sub_elem_2d_broadcast_scalar.mlir [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
```
Comment (linay-xsj) [inline test/Integration/Dialect/TOSA/bf16xbf16_sub_elem_2d_broadcast_scalar/bf16xbf16_sub_elem_2d_broadcast_scalar.mlir:1]: Could you please upload the dut.cc file for a reference?

File: test/Integration/Dialect/TOSA/bf16xbf16_sub_elem_2d_broadcast_2d/bf16xbf16_sub_elem_2d_broadcast_2d.mlir [hunk 3]
```diff
@@ -1,7 +1,6 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
```
Comment (linay-xsj) [inline test/Integration/Dialect/TOSA/bf16xbf16_sub_elem_2d_broadcast_2d/bf16xbf16_sub_elem_2d_broadcast_2d.mlir:1]: Miss defines.h and testbench.cc files.

File: test/Integration/Dialect/TOSA/bf16xbf16_sub_elem_2d_broadcast_scalar/testbench.cc [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    printf("TEST PASSED\n");
+  else
+    printf("TEST FAILED\n");
+
+  return ok ? 0 : 1;
+}
+
+void dut_ref(bfloat16 *in0, bfloat16 *in1, bfloat16 *out0) {
+  for (unsigned k = 0; k < OUT0_SIZE; k += 1) {
+    out0[k] = in0[k] - in1[k % IN1_SIZE];
```
Comment (linay-xsj) [inline test/Integration/Dialect/TOSA/bf16xbf16_sub_elem_2d_broadcast_scalar/testbench.cc:54]: Why do you need k % IN1_SIZE here?

File: lib/Dialect/AIEVec/Transforms/VectorToAIEVecConversions.cpp [hunk 5]
```diff
@@ -1025,12 +1025,6 @@ struct LowerVectorAddOrSubOpToAIEVecAddElemOrSubElemOp
           unsigned rBitWidth =
               rSrcType.getElementType().getIntOrFloatBitWidth();
```
Comment (linay-xsj) [inline lib/Dialect/AIEVec/Transforms/VectorToAIEVecConversions.cpp:1027]: Line 1023-1026 can be deleted.

Comment (jamestcl-amd) [reply lib/Dialect/AIEVec/Transforms/VectorToAIEVecConversions.cpp:1020]: done and get rid of the compilation warning.

File: test/Integration/Dialect/TOSA/bf16xbf16_sub_elem_2d_broadcast_2d/bf16xbf16_sub_elem_2d_broadcast_2d.mlir (hunk 3 above)
Comment (jamestcl-amd) [reply test/Integration/Dialect/TOSA/bf16xbf16_sub_elem_2d_broadcast_2d/bf16xbf16_sub_elem_2d_broadcast_2d.mlir:1]: This is to fix the XFAIL test, so only adding the dut.cc for reference. The defines.h and testbench.cc are already in there.

File: test/Integration/Dialect/TOSA/bf16xbf16_sub_elem_2d_broadcast_scalar/bf16xbf16_sub_elem_2d_broadcast_scalar.mlir (hunk 2 above)
Comment (jamestcl-amd) [reply test/Integration/Dialect/TOSA/bf16xbf16_sub_elem_2d_broadcast_scalar/bf16xbf16_sub_elem_2d_broadcast_scalar.mlir:1]: This is a XFAIL test, so no dut.cc at the moment. Will add the dut.cc when this test is fixed.

File: test/Integration/Dialect/TOSA/bf16xbf16_sub_elem_2d_broadcast_scalar/testbench.cc (hunk 4 above)
Comment (jamestcl-amd) [reply test/Integration/Dialect/TOSA/bf16xbf16_sub_elem_2d_broadcast_scalar/testbench.cc:54]: See defines.h, `IN0_SIZE =  16 * 1024` and `IN1_SIZE = 1`.

File: test/Integration/Dialect/TOSA/i8xi16_sub_elem/dut.cc (hunk 1 above)
Comment (jamestcl-amd) [reply test/Integration/Dialect/TOSA/i8xi16_sub_elem/dut.cc:2]: Similar to the bf16xbf16_sub_elem_2d_broadcast_2d test. This is to fix a XFAIL test, so only adding the dut.cc for reference.

File: test/Integration/Dialect/TOSA/bf16xbf16_sub_elem_2d_broadcast_scalar/testbench.cc (hunk 4 above)
Comment (linay-xsj) [reply test/Integration/Dialect/TOSA/bf16xbf16_sub_elem_2d_broadcast_scalar/testbench.cc:54]: I meant you can directly use in1[0].

**Review by linay-xsj — APPROVED** [review]

**Review by david-vc — APPROVED** [review]

**Pull request merged**